src/chatlog/chatmessage.cpp

```diff
@@ -175,6 +178,58 @@ void ChatMessage::hideDate()
         c->hide();
 }
 
+QString ChatMessage::detectMarkdown(const QString &str)
+{
+    QString out;
+
+    // Create regex for certain markdown syntax
+    QRegExp exp("(\\*\\*)([^\\*\\*]*)(\\*\\*)"   // Bold    **text**
+                "|(\\*)([^\\*]*)(\\*)"           // Italics *text*
+                "|(\\_)([^\\_]*)(\\_)"           // Italics _text_
+                "|(\\_\\_)([^\\_\\_]*)(\\_\\_)"  // Italics __text__
+                "|(\\-)([^\\-]*)(\\-)"           // Underline  -text-
+                "|(\\~)([^\\~]*)(\\~)"           // Strike  ~text~
+                "|(\\~~)([^\\~\\~]*)(\\~~)"      // Strike  ~~text~~
+                );
+
+    // Support for multi-line text
+    QStringList messageLines = str.split("\n");
+    QStringList outLines;
+    for (int i = 0; i < messageLines.size(); ++i)
+    {
+        out = messageLines.at(i);
+        int offset = 0;
+        while ((offset = exp.indexIn(out, offset)) != -1)
+        {
+            QString snippet = exp.cap();
+
+            QString htmledSnippet;
+
+            // Match captured string to corresponding md format
+            if (exp.cap(1) == "**") // Bold **text**
+                htmledSnippet = QString("<b>%1</b>").arg(snippet.mid(2,snippet.length()-4));
+            else if (exp.cap(4) == "*" && snippet.length() > 2) // Italics *text*
+                htmledSnippet = QString("<i>%1</i>").arg(snippet.mid(1,snippet.length()-2));
+            else if (exp.cap(7) == "_" && snippet.length() > 2) // Italics _text_
+                htmledSnippet = QString("<i>%1</i>").arg(snippet.mid(1,snippet.length()-2));
+            else if (exp.cap(10) == "__"&& snippet.length() > 4) // Italics __text__
+                htmledSnippet = QString("<i>%1</i>").arg(snippet.mid(2,snippet.length()-4));
```

Comment:
@agilob What other pattern for bold should be added? :P

It'll have to be adapted to the STS' spec eventually anyways. It's pretty simply to edit/add new syntaxes though.

On January 21, 2016 5:04:32 AM PST, agilob <notifications@github.com> wrote:
>> +        int offset = 0;
>> +        while ((offset = exp.indexIn(out, offset)) != -1)
>> +        {
>> +            QString snippet = exp.cap();
>> +
>> +            QString htmledSnippet;
>> +
>> +            // Match captured string to corresponding md format
>> +            if (exp.cap(1) == "**") // Bold **text**
>> +                htmledSnippet =
>QString("<b>%1</b>").arg(snippet.mid(2,snippet.length()-4));
>> +            else if (exp.cap(4) == "*" && snippet.length() > 2) //
>Italics *text*
>> +                htmledSnippet =
>QString("<i>%1</i>").arg(snippet.mid(1,snippet.length()-2));
>> +            else if (exp.cap(7) == "_" && snippet.length() > 2) //
>Italics _text_
>> +                htmledSnippet =
>QString("<i>%1</i>").arg(snippet.mid(1,snippet.length()-2));
>> +            else if (exp.cap(10) == "__"&& snippet.length() > 4) //
>Italics __text__
>> +                htmledSnippet =
>QString("<i>%1</i>").arg(snippet.mid(2,snippet.length()-4));
>
>Italics 3 times but bold once?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/tux3/qTox/pull/2832/files#r50397534